src/main/java/in/projecteka/gateway/common/Validator.java

```diff
@@ -29,50 +32,45 @@
     CMRegistry cmRegistry;
     CacheAdapter<String, String> requestIdMappings;
 
-    public Mono<ValidatedRequest> validateRequest(HttpEntity<String> requestEntity, String routingKey) {
-        String clientId = requestEntity.getHeaders().getFirst(routingKey);
+    public Mono<ValidatedRequest> validateRequest(HttpEntity<String> maybeRequest, String routingKey) {
```

Comment:
inspite of whether the request is valid or not, its still a request. So why the name "maybeRequest"

Reply:
You are right about the it's still a request part, my thought process was.the request was intended to target system (HIP/HIU/CM), gateway is a mediator,Before it says, I accept the request to redirect the request to the target system, it validates, if validation fails it rejects the request, so that's why I called it as `maybeRequest` , and also we use this generic thing `HttpEntity<String>` , so it can be anything.